### PR TITLE
lightgbm SWIG Java wrapper changes needed to add early stopping in mmlspark

### PR DESCRIPTION
**Pull request by imatiach-msft**
native changes required for this PR: https://github.com/Azure/mmlspark/pull/504
resolves issue: https://github.com/Azure/mmlspark/issues/435

### CODE REVIEW

**Comment by guolinke** [comment]
close-reopen for CI

File: swig/lightgbmlib.i [hunk 1]
```diff
@@ -22,6 +24,26 @@
...
+    if (result != 0) {
+      return nullptr;
+    }
+    return dst;
+  }
+
+  char ** LGBM_BoosterGetEvalNamesSWIG(BoosterHandle handle,
+				       int eval_counts) {
+    char** dst = new char*[eval_counts];
+    for (int i = 0; i < eval_counts; i++) {
```
Comment (StrikerRUS) [inline swig/lightgbmlib.i]: `++i` for the consistency with cpp project's codebase?

Comment (StrikerRUS) [inline swig/lightgbmlib.i]: `++i` for the consistency with cpp LightGBM's codebase?

File: swig/lightgbmlib.i [hunk 2]
```diff
@@ -22,6 +24,26 @@
...
+      result = LGBM_BoosterSaveModelToString(handle, start_iteration, num_iteration, realloc_len, out_len, dst);
+    }
+    if (result != 0) {
+      return nullptr;
+    }
+    return dst;
+  }
+
+  char ** LGBM_BoosterGetEvalNamesSWIG(BoosterHandle handle,
+				       int eval_counts) {
```
Comment (StrikerRUS) [inline swig/lightgbmlib.i]: indent

File: swig/lightgbmlib.i (hunk 1 above)
Comment (imatiach-msft) [reply swig/lightgbmlib.i]: done!

File: swig/lightgbmlib.i (hunk 2 above)
Comment (imatiach-msft) [reply swig/lightgbmlib.i]: good catch, done!

**Pull request merged**